fuel-tx/src/transaction.rs

```diff
@@ -589,6 +597,30 @@ pub mod field {
         fn tx_pointer_static() -> usize;
     }
 
+    pub trait InputContract {
+        fn input_contract(&self) -> &input::contract::Contract;
+        fn input_contract_mut(&mut self) -> &mut input::contract::Contract;
+        fn input_contract_offset(&self) -> usize;
+    }
+
+    pub trait OutputContract {
+        fn output_contract(&self) -> &output::contract::Contract;
+        fn output_contract_mut(&mut self) -> &mut output::contract::Contract;
+        fn output_contract_offset(&self) -> usize;
+    }
+
+    pub trait MintAmount {
+        fn mint_amount(&self) -> &Word;
+        fn mint_amount_mut(&mut self) -> &mut Word;
+        fn mint_amount_offset(&self) -> usize;
+    }
+
+    pub trait MintAssetId {
+        fn mint_asset_id(&self) -> &AssetId;
+        fn mint_asset_id_mut(&mut self) -> &mut AssetId;
+        fn mint_asset_id_offset(&self) -> usize;
```

Comment:
What are the offsets for? It looks like you're calculating the positions of data in a buffer.

Reply:
Yes, it is the index of where the field starts in the canonical byte representation of the transaction.